ignite/distributed/utils.py

```diff
@@ -0,0 +1,220 @@
+        return 0
+
+    @staticmethod
+    def get_world_size() -> int:
+        return 0
+
+    @staticmethod
+    def is_distributed() -> bool:
+        return False
+
+    @staticmethod
+    def name() -> str:
+        return "sequential"
+
+
+class XlaModel:
+    @staticmethod
+    def get_local_rank() -> int:
+        raise Exception("Not yet implemented")
+
+    @staticmethod
+    def get_rank() -> int:
+        return xm.get_ordinal()
+
+    @staticmethod
+    def get_world_size() -> int:
+        return xm.xrt_world_size()
+
+    @staticmethod
+    def is_distributed() -> bool:
+        return has_xla_support and xm.xrt_world_size() > 1
+
+    @staticmethod
+    def name() -> str:
+        return "xla"
+
+
+class DistModel:
+    def __init__(self):
+        if "SLURM_JOB_ID" in os.environ:
+            os.environ["RANK"] = os.environ["SLURM_PROCID"]
+            os.environ["LOCAL_RANK"] = os.environ["SLURM_LOCALID"]
+            os.environ["WORLD_SIZE"] = os.environ["SLURM_NTASKS"]
+            # port should be the same over all process
+            slurm_port = os.environ["SLURM_JOB_ID"]
+            slurm_port = slurm_port[-4:]
+            os.environ["MASTER_PORT"] = str(int(slurm_port) + 15000)
+            # master address is the first hostname of nodes list
+            hostnames = subprocess.check_output(["scontrol", "show", "hostnames", os.environ["SLURM_JOB_NODELIST"]])
+            os.environ["MASTER_ADDR"] = hostnames.split()[0].decode("utf-8")
+        else:
+            os.environ["RANK"] = os.environ.get("RANK", "0")
+            os.environ["LOCAL_RANK"] = os.environ.get("LOCAL_RANK", "0")
+            os.environ["WORLD_SIZE"] = os.environ.get("WORLD_SIZE", "1")
+            os.environ["MASTER_ADDR"] = os.environ.get("MASTER_ADDR", "127.0.0.1")
+            os.environ["MASTER_PORT"] = os.environ.get("MASTER_PORT", "15000")
+        self._master_port = int(os.environ["MASTER_PORT"])
+        # for debug
+        self._master_addr = os.environ["MASTER_ADDR"]
+        self._local_rank = int(os.environ["LOCAL_RANK"])
+
+    def get_local_rank(self) -> int:
+        assert dist.is_available() and dist.is_initialized()
+        return self._local_rank
+
+    @staticmethod
+    def get_rank() -> int:
+        assert dist.is_available() and dist.is_initialized()
+        return dist.get_rank()
+
+    @staticmethod
+    def get_world_size() -> int:
+        assert dist.is_available() and dist.is_initialized()
+        return dist.get_world_size()
+
+    @staticmethod
+    def is_distributed() -> bool:
+        return dist.is_available() and dist.is_initialized() and dist.get_world_size() > 0
+
+    @staticmethod
+    def name() -> str:
+        return "dist"
+
+
+# default is sequential model
+_model = SequentialModel()
+# default backend is None
+_backend = None
+# default device is None
+_device = None
+# the model, device and backend could change at initialization
+
+
+def is_dist_distributed() -> bool:
```

Comment:
Name looks strange. Maybe we need to split by device or backend ?

Reply:
Backend neither device are related to distribution.

There are 2 models : dist or xla. I should prefer to have `is_distributed` and ask to the model (xla or dist). If needed, we can have the model name and discriminate.